packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/representation/services/RepresentationMetadataDetailsViewPageDescriptionProvider.java

```diff
@@ -119,19 +118,18 @@ private TextfieldDescription createLabelTextField() {
                 .map(RepresentationMetadata::getLabel)
                 .orElse("");
 
-        BiFunction<VariableManager, String, IStatus> newValueHandler = (variableManager, newValue) -> {
-            var self = variableManager.get(VariableManager.SELF, RepresentationMetadata.class);
-            if (self.isPresent()) {
-                var representationMetadata = self.get();
-
-                // delegate the renaming of the representation to EditingContextEventProcessor
-                Map<String, Object> parameters = new HashMap<>();
-                parameters.put(EditingContextEventProcessor.REPRESENTATION_ID, representationMetadata.getId().toString());
-                parameters.put(EditingContextEventProcessor.REPRESENTATION_LABEL, newValue);
-                return new Success(ChangeKind.REPRESENTATION_TO_RENAME, parameters);
-            }
-            return new Failure("");
-        };
+        BiFunction<VariableManager, String, IStatus> newValueHandler = (variableManager, newValue) -> variableManager.get(VariableManager.SELF, RepresentationMetadata.class)
+                .map(RepresentationMetadata::getId)
+                .map(representationId -> {
+                    var result = this.representationMetadataUpdateService.updateLabel(null, representationId, newValue);
```

Comment:
Can't we do better than `null`? Because we need the proper value here, we can't afford to lose it.